Assets/Shopify/UIToolkit/Components/ViewSwitcher/ViewSwitcher.cs

```diff
@@ -0,0 +1,129 @@
+﻿namespace Shopify.UIToolkit {
+    using System.Collections;
+    using System.Collections.Generic;
+    using UnityEngine;
+
+    /// <summary>
+    /// A component that allows for switching between multiple UI "views",
+    /// where a view is different UI that needs to be switched between modally.
+    /// 
+    /// In the Unity Buy SDK, we use this for the factory shops in order to switch
+    /// between the cart/product list/product details views in the builtin multi
+    /// product shops.
+    /// </summary>
+    public class ViewSwitcher : MonoBehaviour {
+        public const float AnimationDuration = 0.3f;
```

Comment:
Is the intention to give this to developers to use as well? If so I would make this non-const so devs can change it in the editor